### ISSUE TITLE
Dynamically displayed contents

### ISSUE THREAD

**Issue by mm2t** [w3c/wcag3#804]
Hello.

I have skimmed [WCAG 3.0 Working Draft](https://www.w3.org/TR/wcag-3.0) and could not find anything related to this serious issue. I expected it to be at [§ Structure](https://www.w3.org/TR/wcag-3.0/#structure).

These days, a pervasive form of content presentation on webpages and mobile apps is dynamic structures _for contents_ (not navigation). These basically hide parts and show parts interactively based on user choice. While this is neat for previewing, if a user wants to save the page _contents_ or print the complete contents, great difficulty arises.

-----------

The following are real examples of such dynamic structural elements:

**Expand/Collapse sections which force only one item at time**
<img width="629" height="286" alt="demo image" src="https://github.com/user-attachments/assets/3e564599-6098-4991-83f4-f8a541a9ca6f" />
These are common on product and service marketing pages for things like FAQs. 
If webpages employ transition animations (eg: sliding down box) to display the contents, it can be annoying to users, especially if these animations are slow and the user has to expand/collapse sections often.

**Expand/Collapse sections that allow multiple expanded items**
<img width="629" height="307" alt="demo image" src="https://github.com/user-attachments/assets/43e7a002-aa1c-4601-b8b8-bc7e6275fa32" />
These are very common for FAQs, product & service marketing info pages, product support pages (as in image), etc.
The number of expand/collapse items can be large. Or the number of pages checked can be large.
When saving these pages (as HTML files or screenshot images) to retain the metadata and info, user has to manually expand all sections before saving.

**Content boxes with tabs**
<img width="629" height="298" alt="demo image" src="https://github.com/user-attachments/assets/1ca0f600-e582-4a8b-93f9-8c7a9c3e54c2" />
This is often seen at product info pages, especially (not demonstrated in above example) by vendors that have a large technical inventory (eg: selling electronic components or mechanical parts) where product page has datasheet docs, brochures, application notes, technical drawings, etc.
Again, the user is unable to save a single page with all the info. Sometimes, a box tab might contain just a sentence!

**Tabbed figures**
<img width="629" height="368" alt="demo image" src="https://github.com/user-attachments/assets/5774f43b-fb57-4fff-b8fd-06cba5b128f2" />
These are figures which contain tabs to determine which content gets displayed. The content could be an image, a graph/chart, a mathematical formula, a data table, etc.
Again, when printing or saving the page, only the activated contents is present 😔.

**Before/After image sliders**
<img width="629" height="416" alt="demo image" src="https://github.com/user-attachments/assets/c592f118-0973-42df-af72-3a11f8d79d7b" />
When printing and saving, I would like to see both images: the before and the after. Note that the labels 'Before' and 'After' in the example above only appear when the cursor is hovered over the image (and not in print)!
Things are even worse if I had moved the slider all the way to one side, where it is either the before or after image only seen.

**Image galleries**
<img width="629" height="482" alt="demo image" src="https://github.com/user-attachments/assets/44c222a6-822b-4e95-bc6c-25bdc315e605" />
OK that might be asking for too much. But again, if I want to print a product info page, it makes sense to expect all product images to appear on the printed page and not as they actively are selected on the webpage in the browser.

**More generally: ANY form of dynamic structural elements that display _contents_**
I think the above examples demonstrate the point: all these dynamic elements affect appearance of actual contents. This appearance is reserved even when printing, and there is no convenient way for the user to 'flatten' the page so everything is available for saving.

-----------

This form of UI keeps things simple when _navigating_ and actively interacting with the contents. _However_, they do a horrible job when the user wants to _print_ the contents or save a webpage to retain the information offered.

It is really an agonizing experience for the user to manually expand sections, fiddle with HTML source code to allow multiple sections to be expanded, manually download gallery images, download multiple pages each one showing one tab activate where each tab is just a few sentences long, etc.

-----------

Here are some suggestions for recommendations:
- Enable users to 'expand all' contents persistently. This could be a toggle button at the top or along with light/dark mode controls.
- If not offering the aforementioned, offer a PRINT-FRIENDLY version which has ALL CONTENTS present serially. This should be a webpage I can save normally, apply my own stylesheet overrides over, etc.

<br />

Summary: Dynamic _content_ structural elements are good for readability and neatness when browsing. However, webpages should provide means to access an all-expanded ('flat') static page that presents _all contents_ and not my current viewport state.

Thank you

**Comment by mraccess77** [comment]
On a related note - the low vision task force had proposed in the Low Vision User Requirements a provision for printing customized text. https://www.w3.org/TR/low-vision-needs/#printing-customized-text

**Comment by mm2t** [comment]
> On a related note - the low vision task force had proposed in the Low Vision User Requirements a provision for printing customized text. https://www.w3.org/TR/low-vision-needs/#printing-customized-text

This actually makes the availability of a print-friendly version desirable.
If it itself is a webpage, user can use accessibility style sheets, text-to-speech, etc very easily.

As mentioned, the print-friendly version has all the dynamic structural elements 'flattened'. Thus, all contents is available.
